src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -96,11 +96,18 @@ public override void OnStartActivity(Activity activity, object payload)
                     newOne.SetParentId(ctx.ActivityContext.TraceId, ctx.ActivityContext.SpanId, ctx.ActivityContext.TraceFlags);
                     newOne.TraceStateString = ctx.ActivityContext.TraceState;
 
-                    // Starting the new activity make it the Activity.Current one.
+                    // Start the new activity
                     newOne.Start();
 
+                    if (request.ContentType == "application/grpc" && request.Path.HasValue)
+                    {
+                        newOne.SetTag(GrpcTagHelper.GrpcMethodTagName, request.Path.Value);
+                    }
+
                     // Set IsAllDataRequested to false for the activity created by the framework to only export the sibling activity and not the framework activity
                     activity.IsAllDataRequested = false;
+
+                    // Make the new one Activity.Current
```

Comment:
this comment is misleading. Activity.Current is modified when newOne.Start() is called itself, not at this line.